nixos/roles/nix-remote-builder.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, config, pkgs, ... }:
+let
+  cfg = config.roles.nix-remote-builder;
+in
+{
+  _file = ./nix-remote-builder.nix;
+
+  options.roles.nix-remote-builder = {
+    schedulerPublicKeys = lib.mkOption {
+      description = "SSH public keys of the central build scheduler";
+      type = lib.types.listOf lib.types.str;
+    };
+  };
+
+  config = {
+    # Garbage-collect often
+    nix.gc.automatic = true;
+    nix.gc.dates = "*:45";
+    nix.gc.options = ''--max-freed "$((128 * 1024**3 - 1024 * $(df -P -k /nix/store | tail -n 1 | ${pkgs.gawk}/bin/awk '{ print $4 }')))"'';
```

Comment:
@zimbatm what did you try to calculate here? When I run the command on one of my machines the result is negative.

Reply:
This command ensures that after the GC ran, 128 GB of disk space are free. If your free disk space is already more than 128G, then the calculated number will be nagative, in which case the GC won't delete any paths. This is expected.